shell/platform/fuchsia/flutter/software_surface.cc

```diff
@@ -0,0 +1,404 @@
+  constraints.buffer_memory_constraints.min_size_bytes = 0;
+  constraints.buffer_memory_constraints.max_size_bytes = 0xffffffff;
+  constraints.buffer_memory_constraints.physically_contiguous_required = false;
+  constraints.buffer_memory_constraints.secure_required = false;
+  constraints.buffer_memory_constraints.ram_domain_supported = true;
+  constraints.buffer_memory_constraints.cpu_domain_supported = true;
+  constraints.buffer_memory_constraints.inaccessible_domain_supported = false;
+  constraints.buffer_memory_constraints.heap_permitted_count = 0;
+  constraints.image_format_constraints_count = 1;
+  fuchsia::sysmem::ImageFormatConstraints& image_constraints =
+      constraints.image_format_constraints[0];
+  image_constraints = fuchsia::sysmem::ImageFormatConstraints();
+  image_constraints.min_coded_width = static_cast<uint32_t>(size.fWidth);
+  image_constraints.min_coded_height = static_cast<uint32_t>(size.fHeight);
+  image_constraints.max_coded_width = static_cast<uint32_t>(size.fWidth);
+  image_constraints.max_coded_height = static_cast<uint32_t>(size.fHeight);
+  image_constraints.min_bytes_per_row = 0;
+  image_constraints.max_bytes_per_row = std::numeric_limits<uint32_t>::max();
+  image_constraints.max_coded_width_times_coded_height =
+      std::numeric_limits<uint32_t>::max();
+  image_constraints.pixel_format.type =
+      fuchsia::sysmem::PixelFormatType::R8G8B8A8;
+  image_constraints.color_spaces_count = 1;
+  image_constraints.color_space[0].type = fuchsia::sysmem::ColorSpaceType::SRGB;
+  image_constraints.pixel_format.has_format_modifier = true;
+  image_constraints.pixel_format.format_modifier.value =
+      fuchsia::sysmem::FORMAT_MODIFIER_LINEAR;
+  image_constraints.min_bytes_per_row = static_cast<uint32_t>(size.fWidth) * 4;
+  image_constraints.max_bytes_per_row = static_cast<uint32_t>(size.fWidth) * 4;
+  zx_status_t set_constraints_status =
+      buffer_collection->SetConstraints(true, constraints);
+  if (set_constraints_status != ZX_OK) {
+    FML_LOG(ERROR) << "SoftwareSurface: Failed to set constraints: "
+                   << zx_status_get_string(set_constraints_status);
+    return false;
+  }
+
+  fuchsia::sysmem::BufferCollectionInfo_2 buffer_collection_info;
+  zx_status_t allocation_status = ZX_OK;
+  zx_status_t wait_for_allocated_status =
+      buffer_collection->WaitForBuffersAllocated(&allocation_status,
+                                                 &buffer_collection_info);
+  if (allocation_status != ZX_OK) {
+    FML_LOG(ERROR) << "SoftwareSurface: Failed to allocate: "
+                   << zx_status_get_string(allocation_status);
+    return false;
+  }
+  if (wait_for_allocated_status != ZX_OK) {
+    FML_LOG(ERROR) << "SoftwareSurface: Failed to wait for allocate: "
+                   << zx_status_get_string(wait_for_allocated_status);
+    return false;
+  }
+  if (buffer_collection_info.settings.image_format_constraints.pixel_format
```

Comment:
I dont think these checks are necessary. We can trust sysmem API to fail at WaitForBuffersAllocated() if our constraints arent applied.